### PR TITLE
Airgap upload async

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: api/src/controllers/kots/KotsAPI.ts [hunk 1]
```diff
@@ -380,7 +382,7 @@ export class KotsAPI {
         // parse the dockerconfig secret
         const parsed = JSON.parse(base64Decode(res.body.data[".dockerconfigjson"]));
         const auths = parsed.auths;
-        for (const hostname in auths) {
+        for (const hostname of auths) {
```
Comment (emosbaugh) [inline api/src/controllers/kots/KotsAPI.ts]: @salahalsaleh1993 is this correct?

File: api/src/controllers/kots/KotsAPI.ts [hunk 2]
```diff
@@ -391,7 +393,7 @@ export class KotsAPI {
         }
       }
     } catch {
-      /* no need to handle, rbac problem or not a path we can read registry */
+      // no need to handle, rbac problem or not a path we can read registry
```
Comment (divolgin) [inline api/src/controllers/kots/KotsAPI.ts]: generally not in favor of superfluous refactoring, but this gets a ❤️

Comment (emosbaugh) [reply api/src/controllers/kots/KotsAPI.ts]: i find linting errors very helpful and sometimes can catch bugs. when the file is full of lint issues its easy to ignore them.

File: api/src/controllers/kots/KotsAPI.ts (hunk 1 above)
Comment (salahalsaleh1993) [reply api/src/controllers/kots/KotsAPI.ts]: is auths an object or an array?

Comment (emosbaugh) [reply api/src/controllers/kots/KotsAPI.ts]: must be an object

Comment (salahalsaleh1993) [reply api/src/controllers/kots/KotsAPI.ts]: then you should use `in`

**Pull request merged**